src/nspawn/nspawn-bind-user.c

```diff
@@ -91,19 +91,28 @@ static int convert_user(
                 UserRecord *u,
                 GroupRecord *g,
                 uid_t allocate_uid,
+                const char *shell,
                 UserRecord **ret_converted_user,
                 GroupRecord **ret_converted_group) {
 
         _cleanup_(group_record_unrefp) GroupRecord *converted_group = NULL;
         _cleanup_(user_record_unrefp) UserRecord *converted_user = NULL;
         _cleanup_free_ char *h = NULL;
+        const char *sh = NULL;
         sd_json_variant *p, *hp = NULL, *ssh = NULL;
         int r;
 
         assert(u);
         assert(g);
         assert(user_record_gid(u) == g->gid);
 
+        if (shell) {
+                if (streq(shell, "yes"))
+                        sh = u->shell; /* copy the user record shell */
```

Comment:
Now, the string is not copied, hence it is not necessary to introduce a new variable `sh`:
```c
if (streq_ptr(shell, "yes"))
        shell = u->shell
```